sdc/utilities/prange_utils.py

```diff
@@ -54,36 +54,20 @@ def get_pool_size_impl():
     return get_pool_size_impl
 
 
+@sdc_register_jitable
 def get_chunks(size, pool_size=0):
     if pool_size == 0:
         pool_size = get_pool_size()
 
-    chunk_size = (size - 1) // pool_size + 1
+    pool_size = min(pool_size, size)
+    chunk_size = size // pool_size
```

Comment:
Please return the initial formula `(size - 1)//pool_size + 1`

Reply:
This is not working properly.
With an array of 5 elements and 4 threads, chunks are divided into 0-2, 2-4, 4-6, 6-8

Reply:
No, it divided into chunks [0,2), [2,4), [4,5), [5,5). And that's fine.
I think you should add code to discard last chunk in such case

Reply:
In this module or in the function code that use this?

Reply:
In this module. Something like this:
```python
    pool_size = min(pool_size, size)
    chunk_size = (size - 1)//pool_size + 1

    chunks = []
    for i in range(pool_size):
        start = i*chunk_size
        stop = min((i+1)*chunk_size, size)
        if start >= size:
            break
        
        chunks.append(Chunk(start, stop))
```